src/main/java/org/dependencytrack/persistence/VulnerabilityQueryManager.java

```diff
@@ -887,4 +888,49 @@ public void deleteAffectedVersionAttributions(final Vulnerability vulnerability)
         query.setParameters(vulnerability);
         query.deletePersistentAll();
     }
+
+    /**
+     * Binds the two objects together in a corresponding join table.
+     *
+     * @param vulnerability a Vulnerability object
+     * @param tags    a List of Tag objects
+     */
+    @SuppressWarnings("unchecked")
+    public void bind(Vulnerability vulnerability, List<Tag> tags) {
+        final Query<Tag> query = pm.newQuery(Tag.class, "vulnerabilities.contains(:vulnerability)");
+        final List<Tag> currentVulnerabilityTags = (List<Tag>) query.execute(vulnerability);
+        pm.currentTransaction().begin();
+        for (final Tag tag : currentVulnerabilityTags) {
+            if (!tags.contains(tag)) {
+                tag.getVulnerabilities().remove(vulnerability);
+            }
+        }
+        vulnerability.setTags(tags);
+        for (final Tag tag : tags) {
+            final List<Vulnerability> vulnerabilities = tag.getVulnerabilities();
+            if (!vulnerabilities.contains(vulnerability)) {
+                vulnerabilities.add(vulnerability);
+            }
+        }
+        pm.currentTransaction().commit();
+    }
+
+    /**
+     * Returns a paginated result of vulnerabilities by tag.
+     *
+     * @param tag the tag associated with the Vulnerability
+     * @return a List of vulnerabilities that contain the tag
+     */
+    public PaginatedResult getVulnerabilities(final Tag tag) {
+        final PaginatedResult result;
```

Comment:
Nit: You can omit the declaration of `result` and simply `return execute(query, params);`.